iree/hal/local/loaders/legacy_library_loader.cc

```diff
@@ -81,6 +81,13 @@ static iree_status_t iree_hal_dylib_executable_flatbuffer_verify(
                             "executable library_embedded is missing/empty");
   }
 
+  if (iree_DyLibExecutableDef_sanitized_kind_get(executable_def) ==
+          iree_Sanitizer_kAddress &&
+      !__has_feature(address_sanitizer)) {
```

Comment:
Include `iree/base/target_platform.h` and use the `IREE_SANITIZER_ADDRESS` macro instead of has_feature directly (that macro works across compiler types).